services/galley/src/Galley/Intra/Push.hs

```diff
@@ -171,6 +176,13 @@ push ps = do
     toRecipient p r =
       Gundeck.recipient (_recipientUserId r) (_pushRoute p)
         & Gundeck.recipientClients .~ _recipientClients r
+    -- Ensure that under no circumstances we exceed the threshold
+    removeIfLargeFanout limit =
+      filter
+        ( \p ->
+            (pushRecipientListType p == Data.ListComplete)
+              && (length (pushRecipients p) <= (fromIntegral $ fromRange limit))
```

Comment:
I was under the impression that we don't want to fan out to large teams because lookup of users is expensive. But if it reached this function, I think it is fine for it to go, unless I am missing something. So, if it is only the lookup that we want to avoid, we shouldn't enforce this here.

Reply:
What do you mean exactly? The fan out to a _very_ large audience is also problematic with the current architecture. The idea was to have a single place where we make this decision on what to do with the event. In the near future, I can imagine this being write to a team queue, for instance (instead of using a fanout to all team users).

Reply:
Can you explain why this is problematic? Would it be a problem for galley in terms of too many writes to C*?